core/tests/protractor_utils/ExplorationEditorFeedbackTab.js

```diff
@@ -0,0 +1,109 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the exploration editor's feedback tab, for
+ * use in Protractor tests.
+ */
+
+var general = require('./general.js');
+var until = protractor.ExpectedConditions;
+
+var ExplorationEditorFeedbackTab = function () {
+  /*
+   * Interactive elements
+   */
+  var explorationFeedbackSubject = element(
+    by.css('.protractor-test-exploration-feedback-subject'));
+  var feedbackTabRow = element(
+    by.css('.protractor-test-oppia-feedback-tab-row'));
+  var explorationFeedback = element(
+    by.css('.protractor-test-exploration-feedback'));
+  var feedbackBackButton = element(
+    by.css('.protractor-test-oppia-feedback-back-button'));
+  var feedbackResponseTextArea = element(
+    by.css('.protractor-test-feedback-response-textarea'));
+  var suggestionRowClassName = '.protractor-test-oppia-feedback-tab-row';
+  /*
+   * Buttons
+   */
+  var acceptSuggestionButton = element(
+    by.css('.protractor-test-exploration-accept-suggestion-btn'));
+  var feedbackSendResponseButton = element(
+    by.css('.protractor-test-oppia-feedback-response-send-btn'));
+  var viewSuggestionButton = element(
+    by.css('.protractor-test-view-suggestion-btn'));
+  /*
+   * Workflows
+   */
+  this.acceptSuggestion = function(suggestionDescription) {
+    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
+      var matchingSuggestionRows = rows.filter(function() {
+        return explorationFeedbackSubject.getText().then(function(subject) {
+          return suggestionDescription.indexOf(subject) !== -1;
+        });
+      });
+      expect(matchingSuggestionRows[0].isDisplayed()).toBe(true);
+      matchingSuggestionRows[0].click();
+      expect(viewSuggestionButton.isDisplayed()).toBe(true);
+      viewSuggestionButton.click();
+      expect(acceptSuggestionButton.isDisplayed()).toBe(true);
+      acceptSuggestionButton.click();
+      browser.wait(until.invisibilityOf(acceptSuggestionButton), 5000);
+    });
+  };
+
+  this.expectToHaveFeedbackThread = function() {
+    expect(feedbackTabRow.isPresent()).toBe(true);
+  };
+
+  this.getSuggestionThreads = function() {
+    var threads = [];
+    browser.wait(until.visibilityOf(
+      element.all(by.css(suggestionRowClassName)).first()), 5000);
+    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
+      rows.forEach(function() {
+        explorationFeedbackSubject.getText().then(function(subject) {
+          threads.push(subject);
+        });
+      });
+      return threads;
+    });
+  };
+
+  this.readFeedbackMessages = function() {
+    var messages = [];
+    browser.wait(until.visibilityOf(
```

Comment:
Just to confirm -- in a future PR we are abstracting this away to general.waitUntilVisible() or waitFor.elementToBeVisible() or something like that, right? Would be nice to not repeat "5000" all over the place..

Also, if we're doing that, might be good to file an issue for this to keep track.

Reply:
Yes! I think we kinda got stuck on the naming functions thing in the email thread. This deserves its own PR.